src/invariant/InvariantManagerImpl.cpp

```diff
@@ -35,12 +35,15 @@ namespace stellar
 std::unique_ptr<InvariantManager>
 InvariantManager::create(Application& app)
 {
-    return std::make_unique<InvariantManagerImpl>(app.getMetrics());
+    return std::make_unique<InvariantManagerImpl>(app.getMetrics(),
+                                                  app.getAppConnector());
```

Comment:
Could we just pass a copy of the config instead of the whole app connector? (or perhaps, if we remove the extra assert, even the config is not needed)

Reply:
No, we  use the app connector for reading the config, but we also need to use it to post back to the main thread from the snapshot worker thread when raising an error.

Reply:
posting on main isn't needed